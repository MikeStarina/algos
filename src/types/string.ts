import { ElementStates } from "./element-states";


export type TLettersArray = {
    value: string;
    itemState: ElementStates | undefined;
}