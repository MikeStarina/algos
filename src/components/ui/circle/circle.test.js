import React from "react";
import { Circle } from "./circle.tsx";
import TestRenderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { parseConfigFileTextToJson } from "typescript";
import { ElementStates } from "../../../types/element-states";




describe('Circle component render correct', () => {


    it('Plain Circle render correct', () => {
        const circle = TestRenderer
        .create(<Circle />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Circle w letter render correct', () => {
        const circle = TestRenderer
        .create(<Circle letter={'A'} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Circle w react-elm head prop render correct', () => {
        const circle = TestRenderer
        .create(<Circle head={<Circle />} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Circle w tail prop render correct', () => {
        const circle = TestRenderer
        .create(<Circle tail={'tail'} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Circle w react-elm tail prop render correct', () => {
        const circle = TestRenderer
        .create(<Circle tail={<Circle />} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Circle w react-elm tail prop render correct', () => {
        const circle = TestRenderer
        .create(<Circle tail={<Circle />} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Circle w index prop render correct', () => {
        const circle = TestRenderer
        .create(<Circle index={0} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Small circle render correct', () => {
        const circle = TestRenderer
        .create(<Circle isSmall={true} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Default state Circle render correct', () => {
        const circle = TestRenderer
        .create(<Circle state={ElementStates.Default} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Changing state Circle render correct', () => {
        const circle = TestRenderer
        .create(<Circle state={ElementStates.Changing} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

    it('Modified state Circle render correct', () => {
        const circle = TestRenderer
        .create(<Circle state={ElementStates.Modified} />)
        .toJSON();

        expect(circle).toMatchSnapshot();
    })

})