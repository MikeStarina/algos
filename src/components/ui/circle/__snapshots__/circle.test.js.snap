// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component render correct Changing state Circle render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   changing"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component render correct Circle w index prop render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  >
    0
  </p>
  <div
    className="text text_type_input text_color_input mt-4 absolute tail60 element"
  />
</div>
`;

exports[`Circle component render correct Circle w letter render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    >
      A
    </p>
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component render correct Circle w react-elm head prop render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component render correct Circle w react-elm tail prop render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component render correct Circle w react-elm tail prop render correct 2`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  >
    <div
      className="content "
    >
      <div
        className="text text_type_input text_color_input mb-4 absolute head element"
      />
      <div
        className="circle   default"
      >
        <p
          className="text text_type_circle text_color_input letter"
        />
      </div>
      <p
        className="text text_type_input text_color_input mt-4 absolute index"
      />
      <div
        className="text text_type_input text_color_input mt-4 absolute tail30 element"
      />
    </div>
  </div>
</div>
`;

exports[`Circle component render correct Circle w tail prop render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 string"
  >
    tail
  </div>
</div>
`;

exports[`Circle component render correct Default state Circle render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component render correct Modified state Circle render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   modified"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component render correct Plain Circle render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle   default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;

exports[`Circle component render correct Small circle render correct 1`] = `
<div
  className="content "
>
  <div
    className="text text_type_input text_color_input mb-4 absolute head element"
  />
  <div
    className="circle  small default"
  >
    <p
      className="text text_type_circle text_color_input letter"
    />
  </div>
  <p
    className="text text_type_input text_color_input mt-4 absolute index"
  />
  <div
    className="text text_type_input text_color_input mt-4 absolute tail30 element"
  />
</div>
`;
